### DISCUSSION TITLE
Content rating update job has updated the rating in neo4j as well as redis, elastic search as individual content level but the same was not updated in cassandra content hierarchy

### DISCUSSION THREAD

**Discussion by kamal-singhal** [Sunbird-Knowlg/Community#173] [Bugs]
Hi Team,

Content rating update job has updated the rating in neo4j as well as redis, elastic search as individual content level but the same not updated in Cassandra content hierarchy.
Can you please help let us there are any job that will synch the data neo4j to Cassandra.
can you share any document or KT video where we can understand the complete workflow of content rating update in database and how to come on the UI level.

**Comment by pallakartheekreddy** [comment]
@kamal-singhal Content rating update job will not update in Cassandra (hierarchy data) this is expected behaviour only because hierarchy update is a costly operation, and it will impact performance. 
Will check and update on the document.

**Reply by kamal-singhal** [reply]
@pallakartheekreddy  Thanks for the response. 
Can you tell us following classification on below points:
1. if content rating is not update on the Cassandra (hierarchy data) then how to show the update content rating on the UI because on the UI content rating is being fetching through hierarchy API.
2.  We have checked for hierarchy data of some collection ids that created before migration of diksha, content ratings are present in Cassandra. Can you tell us how to be sync content rating with Cassandra (hierarchy data). there are any specific synch job that will update hierarchy data.  
3. How was it working earlier, presently rating is not getting updated. is there any job that's need to be configured.

**Reply by pallakartheekreddy** [reply]
@kamal-singhal Please find below the answers to your queries.

1. In the UI, the Portal will initially call the hierarchy API to display the list of content. However, when attempting to play individual content, the portal will make a content read API call.
2. Whenever a collection is published, the publish job ensures the retrieval of the latest version of content and stores it in Cassandra. This is how some of the content may have rating data in Cassandra.
3. Yes, we have a Spark job for updating ratings at the content level. Please refer to the discussion thread below for more details.
https://github.com/orgs/Sunbird-Obsrv/discussions/140

**Reply by kamal-singhal** [reply]
@pallakartheekreddy Thanks for the response.

We need to update the content rating on Cassandra, should we republish the collection content it will be updated in Cassandra. Pls confirm.

**Reply by kamal-singhal** [reply]
@pallakartheekreddy  awaiting your response.

**Reply by pallakartheekreddy** [reply]
@kamal-singhal Collection publishing is not a suitable solution because the content rating job updates rating information daily. Therefore, opting for the collection publish approach would require daily publishing after the completion of the rating updater job.

**Reply by kamal-singhal** [reply]
@pallakartheekreddy Thanks for this understanding. We understood the content rating job workflow but not understand before migration of Diksha how was it working. Is there any job that's publish the collection on daily basis after the completion of the rating job.

There are not feasible solution to publishing the collection on daily basis through manually.

**Reply by pallakartheekreddy** [reply]
@kamal-singhal Not sure what do you mean by `before migration of Diksha how was it working`? Can you please help me to understand this step.

**Reply by pallakartheekreddy** [reply]
@kamal-singhal As we haven't received a response from you, we will proceed to close this thread. If you have any further questions or require assistance in the future, please don't hesitate to start a new thread. Thank you.
